python/tk_multi_loader/dialog.py

```diff
@@ -1265,6 +1280,94 @@ def _load_entity_presets(self):
         # data has properly arrived in the model.
         self._on_home_clicked()
 
+    def _setup_hierarchy_model(self, app, root):
+        """
+        Create the model and proxy model required by a hierarchy type configuration setting.
+
+        :param app: :class:`Application`, :class:`Engine` or :class:`Framework` bundle instance
+                    associated with the loader.
+        :param root: The path to the root of the Shotgun hierarchy to display.
+        :return: Created `(model, proxy model)`.
+        """
+
+        # Resolve any magic tokens in the root.
+        if app.context.project:
+            project_id = app.context.project.get("id")
+            root = root.replace("{context.project.id}", str(project_id))
+        else:
+            root = None
+
+        # Construct the hierarchy model.
+        model = SgHierarchyModel(self, bg_task_manager=self._task_manager)
+
+        # Create a proxy model.
+        proxy_model = QtGui.QSortFilterProxyModel(self)
+        proxy_model.setSourceModel(model)
+
+        # Impose and keep the sorting order on the default display role text.
+        proxy_model.sort(0)
+        proxy_model.setDynamicSortFilter(True)
+
+        # We need to provide a dictionary that identifies what additional fields to include
+        # for the loaded hierarchy leaf entities in addition to "id" and "type".
+        # When available, these fields will be used to add info in the detail panel.
+        # TODO: Our entity type list for entities with publishes should be retrieved from the settings.
+        entity_fields = {}
+        for entity_type in ["Asset", "Shot"]:
```

Comment:
I'm wondering if it would be possible (and in if so, how slow would it be) to use the python api's `summarize` method to figure out the various types of entities that `PublishedFile`s on the project/site are linked to. I haven't used `summarize` much at all, so this could be a silly comment. But if that were possible, then you could use the schema to check those entity type's fields and make some educated guesses (with fallbacks) as to what fields should be returned for each type.